### ISSUE TITLE
TransactionBuilder doesn't select the targetCoin in the fastest path

### ISSUE THREAD

**Issue by lontivero** [NicolasDorier/NBitcoin#98]
@NicolasDorier 

In the TransactionBuilder class, the Select method performs a reference comparison and then the targetCoin object is always null. You can see this behaviour running the **CanSelectCoin** test. 

```c#
		public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
		{
			var zero = target.Sub(target);
			var targetCoin = coins
							.FirstOrDefault(c => c.Amount == target);   //<---- HERE
```

```c#
public void CanSelectCoin()
		{
			var selector = new DefaultCoinSelector(0);
			Assert.NotNull(selector.Select(new ICoin[] { CreateCoin("10.0") }, Money.Parse("10.0")));
```

**Comment by lontivero** [comment]
It should use the CompareTo method.

```c# 
			var targetCoin = coins
							.FirstOrDefault(c => c.Amount.CompareTo(target) == 0);
```

If you want, I can fix it. I have a branch with this and some other minor changes.

**Comment by NicolasDorier** [comment]
thanks, fixed in b8982c81
